hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MultiVersionConcurrencyControl.java

```diff
@@ -140,7 +146,7 @@ public WriteEntry begin() {
   public WriteEntry begin(Runnable action) {
     synchronized (writeQueue) {
       long nextWriteNumber = writePoint.incrementAndGet();
-      WriteEntry e = new WriteEntry(nextWriteNumber);
+      WriteEntry e = cachedWriteEntries.get().setWriteNumber(nextWriteNumber).markCompleted(false);
```

Comment:
Should we have protection here in case we overwrite an on-going WriteEntry? One that is not complete?

Reply:
after change `WriteEntry` to threadlocal.
It is ok to this scene in one thread: `mvcc1(begin)---mvcc1(wait complete)---mvcc2(begin)---mvcc2(wait complete)`
but is bad to the scene in one thread: `mvcc1(begin)---mvcc2(begin)---mvcc1(wait complete)---mvcc2(wait complete)`
I think our usage of `WriteEntry` is the former